lib/src/main/java/io/ably/lib/realtime/ChannelBase.java

```diff
@@ -627,21 +610,9 @@ public void run() {
     }
 
     /* State changes provoked by ConnectionManager state changes. */
-
-    public void setConnected(boolean reattachOnResumeFailure) {
-        if (reattachOnResumeFailure && state.isReattachable()){
-            attach(true,null);
-        } else if (state == ChannelState.suspended) {
-            /* (RTL3d) If the connection state enters the CONNECTED state, then
-             * a SUSPENDED channel will initiate an attach operation. If the
-             * attach operation for the channel times out and the channel
-             * returns to the SUSPENDED state (see #RTL4f)
-             */
-            try {
-                attachWithTimeout(null);
-            } catch (AblyException e) {
-                Log.e(TAG, "setConnected(): Unable to initiate attach; channel = " + name, e);
-            }
+    public void setConnected() {
```

Comment:
Could you please explain why we changed method signature?

Reply:
Btw, I wanted to add that there was a discussion for `ably-java` where lots of methods are marked public which are not necessarily used.

Reply:
Because we don't need `reattachOnResumeFailure`, new spec explicitly mentions to attach channels irrespective of resume success/failure.

Reply:
Can we mark old method as deprecated as well? I don't want to introduce breaking changes if we can avoid them

Reply:
As I mentioned => https://github.com/ably/ably-java/pull/982#discussion_r1431570929, this is not being used as public. If someone is using it as a public method, then they are tampering with library internals. I think we should force users not to use this method rather than deprecating it. They should create an explicit issue if they want a new functionality instead of using library internals.

Reply:
I feel there's a plan as a part of `java-2.0` release, where we refactor code to remove deprecated public methods and mark public methods as private which are not part of the spec.

Reply:
Also related to comment -> https://ably-real-time.slack.com/archives/CTQELK4NM/p1612514956037700?thread_ts=1612354862.033900&cid=CTQELK4NM

Reply:
So, it's not just method name, rather implementation can also change for internal methods. So, it's not recommended to use internal methods. I think we should mark those private as such wherever possible.